test/Libraries/Microsoft.Extensions.Caching.Hybrid.Tests/ServiceConstructionTests.cs

```diff
@@ -81,6 +296,107 @@ public void CanParseOptions_WithEntryOptions() // in particular, check we can pa
         Assert.Equal(TimeSpan.FromSeconds(120), defaults.LocalCacheExpiration);
         Assert.Null(defaults.Expiration); // wasn't specified
     }
+
+    [Fact]
+    public void CanCreateKeyedServicesWithKeyedDistributedCaches_UsingNamedOptions()
+    {
+        var source = new JsonConfigurationSource { Path = "BasicConfig.json" };
+        var configBuilder = new ConfigurationBuilder { Sources = { source } };
+        var config = configBuilder.Build();
+
+        var services = new ServiceCollection();
+        services.AddKeyedSingleton<IDistributedCache, CustomMemoryDistributedCache1>("DistributedOne");
+        services.AddKeyedSingleton<IDistributedCache, CustomMemoryDistributedCache2>("DistributedTwo");
+        services.AddOptions<HybridCacheOptions>("HybridOne").Configure(options => ConfigurationBinder.Bind(config, "HybridOne", options));
+        services.AddOptions<HybridCacheOptions>("HybridTwo").Configure(options => ConfigurationBinder.Bind(config, "HybridTwo", options));
+        services.AddKeyedHybridCache(typeof(CustomMemoryDistributedCache1), "HybridOne");
+        services.AddKeyedHybridCache(typeof(CustomMemoryDistributedCache2), "HybridTwo");
+
+        using ServiceProvider provider = services.BuildServiceProvider();
+        var hybridOne = Assert.IsType<DefaultHybridCache>(provider.GetRequiredKeyedService<HybridCache>(typeof(CustomMemoryDistributedCache1)));
+        var hybridOneOptions = hybridOne.Options;
+        var hybridOneBackend = Assert.IsType<CustomMemoryDistributedCache1>(hybridOne.BackendCache);
+        Assert.Equal("DistributedOne", hybridOneOptions.DistributedCacheServiceKey);
+
+        var hybridTwo = Assert.IsType<DefaultHybridCache>(provider.GetRequiredKeyedService<HybridCache>(typeof(CustomMemoryDistributedCache2)));
+        var hybridTwoOptions = hybridTwo.Options;
+        var hybridTwoBackend = Assert.IsType<CustomMemoryDistributedCache2>(hybridTwo.BackendCache);
+        Assert.Equal("DistributedTwo", hybridTwoOptions.DistributedCacheServiceKey);
+
+        provider.GetRequiredKeyedService<HybridCache>(typeof(CustomMemoryDistributedCache1));
+        provider.GetRequiredKeyedService<HybridCache>(typeof(CustomMemoryDistributedCache1));
+        provider.GetRequiredKeyedService<HybridCache>(typeof(CustomMemoryDistributedCache1));
+
+        provider.GetRequiredKeyedService<HybridCache>(typeof(CustomMemoryDistributedCache2));
+        provider.GetRequiredKeyedService<HybridCache>(typeof(CustomMemoryDistributedCache2));
+        provider.GetRequiredKeyedService<HybridCache>(typeof(CustomMemoryDistributedCache2));
+    }
+
+    [Fact]
+    public void CanCreateKeyedServicesWithKeyedDistributedCaches_UsingSetupActions()
+    {
+        var source = new JsonConfigurationSource { Path = "BasicConfig.json" };
```

Comment:
This made reading the test harder for me as I could not see where `"DistributedOne"` was set in the options. Maybe use an in-memory configuration and define it inline here. I don't think it makes it any easier to use a file based config.

For instance https://github.com/dotnet/aspire/blob/f528ede3c5ed78f73cf49328710dd4ec952e03a1/tests/Aspire.Azure.AI.Inference.Tests/AspireAzureAIInferenceExtensionTests.cs#L48-L50